src/shadowfax.py

```diff
@@ -139,7 +139,7 @@ async def shadowfax(message, guid, logger):
 
     # Find the nodes with most significant co-occurrence
     pruned_curies = []
-    while len(pruned_curies) < CURIE_PRUNING_LIMIT and len(pruned_curies) < len(curies):
+    while len(pruned_curies) < CURIE_PRUNING_LIMIT and len(pruned_curies) < len(curie_info.keys()):
```

Comment:
Some performance things:
- let's move `len(curie_info.keys())` to a variable so we aren't computing it for each iteration.
- It looks like all we're doing here is sorting the curie_info dict and iteratively pushing the top scores into a new list. Why don't we just sort the dict once and then take the top n? Something like this:
`pruned_curies = sorted(curie_info.keys(), key=lambda x: curie_info[x]["score"])[:CURIE_PRUNING_LIMIT]`
You might miss out on some if the scores are all the same, but maybe the pruning limit is sufficiently high that we don't care about those low ones anyways.